drivers/pwm/pwm_mspm0.c

```diff
@@ -0,0 +1,166 @@
+#define IS_TIMA(tim)     (tim == TIMA0 || tim == TIMA1)
+// from the datasheet, only TIMA0 has 4 capture compare channels
+#define HAS_FOUR_CC(tim) (tim == TIMA0)
+
+LOG_MODULE_REGISTER(pwm_mspm0, CONFIG_PWM_LOG_LEVEL);
+
+struct pwm_mspm0_config {
+	GPTIMER_Regs *timer;
+	const struct pinctrl_dev_config *pinctrl;
+	const uint32_t frequency;
+	const DL_Timer_ClockConfig clock_cfg;
+	const DL_TIMER_PWM_MODE pwm_mode;
+	const uint8_t cc_index;
+};
+
+static inline int pwm_mspm0_get_cc_direction(int cc_index, int *direction)
+{
+	switch (cc_index) {
+	case DL_TIMER_CC_0_INDEX:
+		*direction = DL_TIMER_CC0_OUTPUT;
+		return 0;
+	case DL_TIMER_CC_1_INDEX:
+		*direction = DL_TIMER_CC1_OUTPUT;
+		return 0;
+	case DL_TIMER_CC_2_INDEX:
+		*direction = DL_TIMER_CC2_OUTPUT;
+		return 0;
+	case DL_TIMER_CC_3_INDEX:
+		*direction = DL_TIMER_CC3_OUTPUT;
+		return 0;
+	default:
+		LOG_DBG("Error, invalid CC index: %d", cc_index);
+		return -EINVAL;
+	};
+}
+
+static int pwm_mspm0_get_cycles_per_sec(const struct device *dev, uint32_t channel,
+					uint64_t *cycles)
+{
+	const struct pwm_mspm0_config *config = dev->config;
+	*cycles = config->frequency /
+		  ((config->clock_cfg.divideRatio + 1) * (config->clock_cfg.prescale + 1));
+	return 0;
+}
+
+static int pwm_mspm0_set_cycles(const struct device *dev, uint32_t channel, uint32_t period_cycles,
+				uint32_t pulse_cycles, pwm_flags_t flags)
+{
+	const struct pwm_mspm0_config *cfg = dev->config;
+	if (channel != 0) {
+		LOG_DBG("Error, only channel 0 is supported");
+		return -EINVAL;
+	}
+
+	if (flags) {
+		LOG_DBG("Error, pwm polarity is not supported");
+		return -ENOTSUP;
+	}
+
+	if (pulse_cycles == 0) {
+		DL_Timer_stopCounter(cfg->timer);
+		return 0;
+	}
+
+	DL_Timer_startCounter(cfg->timer);
+	DL_Timer_setLoadValue(cfg->timer, period_cycles);
+	DL_Timer_setCaptureCompareValue(cfg->timer, pulse_cycles, cfg->cc_index);
+	return 0;
+}
+
+static const struct pwm_driver_api pwm_mspm0_driver_api = {
+	.set_cycles = pwm_mspm0_set_cycles,
+	.get_cycles_per_sec = pwm_mspm0_get_cycles_per_sec,
+};
+
+static int pwm_mspm0_init(const struct device *dev)
+{
+	const struct pwm_mspm0_config *cfg = dev->config;
+	int ret = pinctrl_apply_state(cfg->pinctrl, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		LOG_ERR("Error, failed to apply pinctrl: %d", ret);
+		return ret;
+	}
+
+	DL_Timer_reset(cfg->timer);
+	DL_Timer_enablePower(cfg->timer);
+	DL_Timer_setClockConfig(cfg->timer, (DL_Timer_ClockConfig *)&cfg->clock_cfg);
+
+	// TIMG/TIMA have different struct and init functions. Also we do not need to store
+	// the configuration for each module
+	if (IS_TIMA(cfg->timer)) {
+		// the period value doesn't matter, but it is a req to setup correctly the pwm
+		// we will eventually set the correct period by calling set_cycles
+		const DL_TimerA_PWMConfig pwm_cfg = {
+			.pwmMode = cfg->pwm_mode,
+			.startTimer = DL_TIMER_STOP,
+			.isTimerWithFourCC = HAS_FOUR_CC(cfg->timer),
+			.period = 1000,
```

Comment:
```suggestion
			.period = 0,
```
1000 seems somewhat thought out. How about just putting `0` as the dummy value?